src/Type/ArrayType.php

```diff
@@ -371,6 +371,23 @@ public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $uni
 
 	public function setExistingOffsetValueType(Type $offsetType, Type $valueType): Type
 	{
+		if ($this->itemType->isConstantArray()->yes() && $valueType->isConstantArray()->yes()) {
+			$constArrays = $valueType->getConstantArrays();
+			$newItemType = $this->itemType;
+			foreach($constArrays as $constArray) {
+				foreach($constArray->getKeyTypes() as $keyType) {
+					$newItemType = $newItemType->setExistingOffsetValueType($keyType, $constArray->getOffsetValueType($keyType));
+				}
+			}
+
+			if ($newItemType !== $this->itemType) {
+				return new self(
+					$this->keyType,
+					$newItemType
+				);
+			}
+		}
```

Comment:
required to prevent regression in 

```
@ -0,0 +1,22 @@
<?php

namespace AssignNestedArrays;

use function PHPStan\debugScope;
use function PHPStan\Testing\assertType;

class Foo
{
	public function doBar(int $i, int $j)
	{
		$array = [];

		$array[$i][$j]['bar'] = 1;
		debugScope();
		$array[$i][$j]['baz'] = 2;
		debugScope();

		assertType('non-empty-array<int, non-empty-array<int, array{bar: 1, baz: 2}>>', $array);
	}

}
```